R/clean_glm_data.R

```diff
@@ -0,0 +1,27 @@
+clean_glm_data <- function(mod, data, weights){
+
+  # Grab only necessary variables
+  data <- get_all_vars(mod, data)
+
+  if(!is.null(weights) & length(weights) != nrow(data))
+    stop('`weights` and `data` must be the same length.')
+
+  # Add weights
```

Comment:
What about:

* weights defaults to `NULL`
* if it's not `NULL`, it has to be the same length as `data` (as we have here)
* if it's not provided, *and* if `mod$prior.weights` isn't all 1, then use the names of `mod$prior.weights` to subset `data`  by row names?